app/services/notify/batch-send-notifications.service.js

```diff
@@ -0,0 +1,101 @@
+'use strict'
+
+/**
+ * Batch send notifications to Notify
+ * @module BatchSendNotificationsService
+ */
+
+const NotifyEmailService = require('./notify-email.service.js')
+const NotifyLetterService = require('./notify-letter.service.js')
+const notifyRateLimit = 3000
+const NotifyEmailPresenter = require('./notify-returns-email.presenter.js')
+const NotifyLetterPresenter = require('./notify-returns-letter.presenter.js')
+
+/**
+ *
+ * @param recipients
+ */
+async function go(recipients) {
+  try {
+    const returnsPeriod = {
+      periodEndDate: '28th January 2025',
+      periodStartDate: '1st January 2025',
+      returnDueDate: '28th April 2025'
+    }
+
+    const referenceCode = 'developer-testing'
+
+    console.time('testTime')
+
+    // Journey -Send to notify
+    const notifications = await _promiseSettled(recipients, returnsPeriod, referenceCode)
+    // Add to the scheduled notification table - with notification id
+
+    // Get status ? - update in notification table
+
+    // Add error handling
+    _notificationErrors(notifications)
+
+    console.timeEnd('testTime')
+
+    // Stripe promise status "settled"
+    return notifications.map((notification) => notification.value)
+  } catch (e) {
+    console.log(e)
+  }
+}
+
+async function _sendLetter(recipient, returnsPeriod, referenceCode) {
+  const formatedData = NotifyLetterPresenter.go(recipient, returnsPeriod, referenceCode)
+
+  return NotifyLetterService.go(formatedData.templateId, formatedData.options).then((notifyData) => {
+    return _scheduledNotification(notifyData, recipient, formatedData)
+  })
+}
+
+async function _sendEmail(recipient, returnsPeriod, referenceCode) {
+  const formatedData = NotifyEmailPresenter.go(recipient, returnsPeriod, referenceCode)
```

Comment:
Doing this to the array prior to this should allow us to handle the errors / success simpler in the batch. 

If we can remove the need to use the contact hash then i think this is better. 

We get all the results resolved promises and we  can handle the errors and resend the 'payload' 

```javascript
  const failed = notifications.filter((response) => response.status !== 'fulfilled')
```